src/Functions/array/arrayFold.cpp

```diff
@@ -32,6 +32,9 @@ class FunctionArrayFold : public IFunction
     size_t getNumberOfArguments() const override { return 0; }
     bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
 
+    /// Avoid the default adaptors since they modify the inputs and that makes knowing the lambda argument types
```

Comment:
Ok, but we can also do the following:
- default implementations are ineffective in the presence of lambda functions as arguments.